lego/apps/events/tasks.py

```diff
@@ -475,7 +477,11 @@ def bump_waiting_users_to_new_pool(self, logger_context=None):
     )
     for event_id in events_ids:
         with transaction.atomic():
-            locked_event = Event.objects.select_for_update().get(pk=event_id)
+            locked_event = (
+                Event.objects.select_related("pools")
```

Comment:
Why is the `select_related("pools")` needed here? My understanding here is that the `of` arg will lock related objects, in this case `pools`, so wouldn't that be sufficient?

Reply:
Oops, yeah I initially used a `select_related` to lock the pools but realized that that won't work, and we have to use the `of` parameter here.

Reply:
I had to change this as well, since we apparently can't lock rows of related objects like this. We can only lock related objects included in the query, which pools are not here. So I instead opted to just lock the event, and then lock all the pools. I _think_ this should be fine, since the order of locking shouldn't matter since we only check the reg counter when the event is merged, and the lock should be acquired in all those places.